src/utils.cpp

```diff
@@ -84,9 +84,12 @@ bool TextChat::serialize(string *d)
     d->append((char*)&ou, sizeof ou);
     d->append((char*)&ts, sizeof(ts));
 
+    char isArchive = archive ? 1 : 0;
+    d->append((char*)&isArchive, 1);
+
```

Comment:
Better to place this new flag after the byte of `hasAttachments` (and before the empty bytes for compatibility), since very likely the _attachment code_ will be deployed before than the _archive code_